server/resp/src/main/java/org/infinispan/server/resp/commands/connection/CLIENT.java

```diff
@@ -0,0 +1,130 @@
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.group.ChannelGroup;
+import io.netty.channel.group.ChannelMatcher;
+
+/**
+ * <a href="https://redis.io/commands/client/">CLIENT</a> *
+ *
+ * @since 15.0
+ */
+public class CLIENT extends RespCommand implements Resp3Command {
+   public CLIENT() {
+      super(-1, 0, 0, 0);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler, ChannelHandlerContext ctx,
+                                                      List<byte[]> arguments) {
+      String subcommand = utf8(arguments.get(0)).toUpperCase();
+      ConnectionMetadata metadata = new ConnectionMetadata(ctx.channel());
+      switch (subcommand) {
+         case "CACHING":
+         case "UNPAUSE":
+         case "PAUSE":
+         case "NO-EVICT":
+         case "KILL":
+         case "NO-TOUCH":
+            ByteBufferUtils.stringToByteBuf("-ERR unsupported command\r\n", handler.allocatorToUse());
+            break;
+         case "SETINFO":
+            for (int i = 1; i < arguments.size(); i++) {
+               String name = utf8(arguments.get(i));
+               switch (name.toUpperCase()) {
+                  case "LIB-NAME":
+                     metadata.clientLibraryName(utf8(arguments.get(++i)));
+                     break;
+                  case "LIB-VER":
+                     metadata.clientLibraryVersion(utf8(arguments.get(++i)));
+                     break;
+                  default:
+                     ByteBufferUtils.stringToByteBuf("-ERR unsupported attribute " + name + "\r\n", handler.allocatorToUse());
+                     return handler.myStage();
+               }
+            }
+            break;
+         case "SETNAME":
+            metadata.clientName(utf8(arguments.get(1)));
+            Consumers.OK_BICONSUMER.accept(null, handler.allocatorToUse());
+            break;
+         case "GETNAME":
+            handleBulkResult(metadata.clientName(), handler.allocatorToUse());
+            break;
+         case "GETREDIR":
+            ByteBufferUtils.stringToByteBuf("-ERR unsupported command\r\n", handler.allocatorToUse());
+            break;
+         case "ID":
+            Consumers.LONG_BICONSUMER.accept(metadata.id(), handler.allocatorToUse());
+            break;
+         case "UNBLOCK":
+            ByteBufferUtils.stringToByteBuf("-ERR unsupported command\r\n", handler.allocatorToUse());
+            break;
+         case "INFO": {
+            StringBuilder sb = new StringBuilder();
+            addInfo(sb, metadata);
+            handleBulkResult(sb, handler.allocatorToUse());
+            break;
+         }
+         case "LIST": {
+            handler.checkPermission(AuthorizationPermission.ADMIN);
+            StringBuilder sb = new StringBuilder();
+            ChannelMatcher matcher = handler.respServer().getChannelMatcher();
+            ChannelGroup channels = handler.respServer().getTransport().getAcceptedChannels();
```

Comment:
Running locally, I am receiving an NPE here.

```text
2023-05-19 11:37:19,411 WARN  [o.i.s.r.RespHandler] ISPN012003: Received an unexpected exception. java.lang.NullPointerException: Cannot invoke "org.infinispan.server.core.transport.NettyTransport.getAcceptedChannels()" because the return value of "org.infinispan.server.resp.RespServer.getTransport()" is null
	at org.infinispan.server.resp.commands.connection.CLIENT.perform(CLIENT.java:90)
```

Reply:
Yeah, I forgot to change that so it can work in single port mode

Reply:
Repushed